src/components/common/ColonyHome/ColonySubscription/ColonySubscription.tsx

```diff
@@ -33,43 +38,76 @@ const MSG = defineMessages({
 });
 
 const ColonySubscription = () => {
-  const { colony } = useColonyContext();
-  const { colonyAddress } = colony || {};
+  const { colony, canInteractWithColony, updateColony } = useColonyContext();
+  const {
+    user,
+    updateUser,
+    wallet,
+    walletConnecting,
+    connectWallet,
+    initConnect,
+  } = useAppContext();
+  const navigate = useNavigate();
 
-  const { user } = useAppContext();
+  const watchedItem = watchingColony(colony, user);
 
-  const isSubscribed = !!(user?.watchlist?.items || []).find(
-    (item) => (item?.colony as Colony)?.colonyAddress === colonyAddress,
-  );
+  /* Watch (follow) a colony */
+  const [watch, { data: watchState, loading: loadingWatch }] =
+    useCreateWatchedColoniesMutation({
+      variables: {
+        input: {
+          colonyID: colony?.colonyAddress || '',
+          userID: user?.walletAddress || '',
+        },
+      },
+    });
+
+  /* Unwatch (unfollow) a colony */
+  const [unwatch, { data: unwatchState, loading: loadingUnwatch }] =
+    useDeleteWatchedColoniesMutation({
+      variables: { input: { id: watchedItem?.id || '' } },
+    });
+
+  /* Update user on watch/unwatch */
+  useEffect(() => {
+    if (updateUser && updateColony) {
+      updateUser(user?.walletAddress);
+      updateColony();
+    }
+  }, [user, updateUser, watchState, unwatchState, updateColony]);
+
+  const handleJoinClick = () => {
+    if (user) {
+      watch();
+    } else if (wallet && !user) {
```

Comment:
There is a modal that we planned to add with the new onboarding for creating a user account without navigating away from a Colony. The design is here - https://www.figma.com/file/xGRNwVyii0qaCkWs4hASUn/Onboarding-2.0?node-id=2208%3A45823&t=s6z7HdeHEKIkptnb-1

I have allowed for it here.